core/tests/test_util_jobs.py

```diff
@@ -66,21 +74,44 @@ class ExpCopiesMRJobManager(
 
     @classmethod
     def _get_continuous_computation_class(cls):
+        """Returns the ExpCopiesAggregator class associated with this MapReduce
+        job.
+        """
         return ExpCopiesAggregator
 
     @classmethod
     def entity_classes_to_map_over(cls):
+        """Returns the ExplorationModel domain object."""
         return [exp_models.ExplorationModel]
 
     @staticmethod
     def map(item):
+        """Yields the string containing exploration id with the copy number of
+        the existing exploration.
+
+        Args:
+            item: An exploration domain object.
+
+        Yields:
+            str. The string containing exploration id with the copy number of
+            the existing exploration. It is of the format:
+                <exp_id>copy<copy_number>
+        """
         if ExpCopiesMRJobManager._entity_created_before_job_queued(item):
             for count in range(10):
                 yield ('%scopy%d' % (item.id, count),
                        exp_services.get_exploration_from_model(item).to_yaml())
 
     @staticmethod
     def reduce(exp_id, list_of_exps):
+        """Saves and publishes the newly created copy of the existing
+        exploration.
+
+        Args:
+            exp_id: str. The exploration id.
+            list_of_exps: list(str). The list containing explorations in the
```

Comment:
I don't understand the relationship between exp_id and list_of_exps...

Reply:
The `exp_id` is the exploration id. `list_of_exps` contains the list of all the copies of the exploration with given id to be saved and published.